### PR TITLE
Crypto

### PR DESCRIPTION
**Pull request by arthurchan35**
Hey @martamedio @csierra 
I think we can use the pr as base to discuss and extend anthing we want to add.

### CODE REVIEW

File: modules/apps/portal-crypto-hash-generator/portal-crypto-hash-generator-message-digest/src/main/java/com/liferay/portal/crypto/hash/generator/message/digest/MessageDigestHashGenerator.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash-generator/portal-crypto-hash-generator-message-digest/src/main/java/com/liferay/portal/crypto/hash/generator/message/digest/MessageDigestHashGenerator.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash-generator/portal-crypto-hash-generator-message-digest/src/main/java/com/liferay/portal/crypto/hash/generator/message/digest/factory/MessageDigestHashGeneratorFactory.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash-generator/portal-crypto-hash-generator-message-digest/src/main/java/com/liferay/portal/crypto/hash/generator/message/digest/factory/MessageDigestHashGeneratorFactory.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/context/builder/HashContextBuilderImpl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/context/builder/HashContextBuilderImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/context/builder/HashGeneratorBuilderImpl.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/context/builder/HashGeneratorBuilderImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/flavor/HashFlavorImpl.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/flavor/HashFlavorImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/context/HashGenerationContextImpl.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/context/HashGenerationContextImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/response/HashGenerationResponseImpl.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/generation/response/HashGenerationResponseImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/processor/HashProcessorImpl.java [hunk 8]
```diff
@@ -0,0 +1,227 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/processor/HashProcessorImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/verification/context/HashVerificationContextImpl.java [hunk 9]
```diff
@@ -0,0 +1,126 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/verification/context/HashVerificationContextImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:sf

**Pull request closed without merging**